From d22fef8620cdd87490754f607e7153979731c69d Mon Sep 17 00:00:00 2001
From: Alexey Sokolov <alexey+znc@asokolov.org>
Date: Fri, 13 Jul 2018 22:50:47 +0100
Subject: [PATCH] Better cleanup lines coming from network.

Thanks for Jeriko One <jeriko.one@gmx.us> for finding and reporting this.
---
 src/Client.cpp  | 3 ++-
 src/IRCSock.cpp | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/Client.cpp b/src/Client.cpp
index cec976c537..c0efd62834 100644
--- a/src/Client.cpp
+++ b/src/Client.cpp
@@ -103,7 +103,8 @@ void CClient::ReadLine(const CString& sData) {
     CLanguageScope user_lang(GetUser() ? GetUser()->GetLanguage() : "");
     CString sLine = sData;
 
-    sLine.TrimRight("\n\r");
+    sLine.Replace("\n", "");
+    sLine.Replace("\r", "");
 
     DEBUG("(" << GetFullName() << ") CLI -> ZNC ["
         << CDebug::Filter(sLine) << "]");
diff --git a/src/IRCSock.cpp b/src/IRCSock.cpp
index e9db5447c5..259881dc84 100644
--- a/src/IRCSock.cpp
+++ b/src/IRCSock.cpp
@@ -156,7 +156,8 @@ void CIRCSock::Quit(const CString& sQuitMsg) {
 void CIRCSock::ReadLine(const CString& sData) {
     CString sLine = sData;
 
-    sLine.TrimRight("\n\r");
+    sLine.Replace("\n", "");
+    sLine.Replace("\r", "");
 
     DEBUG("(" << m_pNetwork->GetUser()->GetUserName() << "/"
               << m_pNetwork->GetName() << ") IRC -> ZNC [" << sLine << "]");
